rust/src/wallet.rs

```diff
@@ -520,19 +521,20 @@ pub async fn open_channel(peer_info: PeerInfo, taker_amount: u64) -> Result<()>
 /// If the first channel is not usable, it might be because we've lost
 /// the connection with the 10101 maker, according to the
 /// `rust-lightning` logs.
-pub fn is_first_channel_usable() -> Result<bool> {
-    let channel_manager = {
-        let lightning = &get_wallet()?.lightning;
-
-        lightning.channel_manager.clone()
-    };
-
-    let is_usable = match channel_manager.list_channels().first() {
+pub fn is_first_channel_usable() -> bool {
+    match get_first_channel_details() {
         Some(channel_details) => channel_details.is_usable,
-        None => return Ok(false),
-    };
+        None => false,
+    }
+}
+
+pub fn get_first_channel_details() -> Option<ChannelDetails> {
+    let channel_manager = match &get_wallet() {
+        Ok(wallet) => Some(wallet.lightning.channel_manager.clone()),
+        Err(_) => None,
+    }?;
 
-    Ok(is_usable)
+    channel_manager.list_channels().first().cloned()
```

Comment:
nit:
I think we actually have a helper function `get_channel_manager()`, which returns a `Result`. With this, we can use `ok()` to convert it to an `Option` easily!

```suggestion
    get_channel_manager().ok().map(|cm| cm.list_channels().first().cloned())
```